### PR TITLE
PAYARA-2707 Dynamic registration of MBean as MP metric

### PR DESCRIPTION
**Pull request by Cousjava**
This allows for placeholder replacement in MP custom metrics

### CODE REVIEW

**Comment by Cousjava** [comment]
Jenkins test please

**Comment by Cousjava** [comment]
Jenkins test please

**Comment by smillidge** [comment]
@Cousjava this now has conflicts can you resolve them

**Comment by Cousjava** [comment]
Jenkins test please

File: appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java [hunk 1]
```diff
@@ -192,7 +204,7 @@ public void resetMetricsSecureProperty() {
         MetricRegistry registry = getRegistry(registryName);
         Map<String, Metric> metricMap = registry.getMetrics();
         if (metricMap.containsKey(metricName)) {
-            return singletonMap(metricName, metricMap.get(metricName));
+            return Collections.singletonMap(metricName, metricMap.get(metricName));
```
Comment (arjantijms) [inline appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java]: Does "Collections.singletonMap" really make the code more readable than "singletonMap"?

File: appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java [hunk 2]
```diff
@@ -202,7 +214,7 @@ public void resetMetricsSecureProperty() {
         MetricRegistry registry = getRegistry(registryName);
         Map<String, Metadata> metricMetadataMap = registry.getMetadata();
         if (metricMetadataMap.containsKey(metricName)) {
-            return singletonMap(metricName, metricMetadataMap.get(metricName));
+            return Collections.singletonMap(metricName, metricMetadataMap.get(metricName));
```
Comment (arjantijms) [inline appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java]: Does "Collections.singletonMap" really make the code more readable than "singletonMap"?

Comment (Cousjava) [reply appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java]: Yes.

Comment (arjantijms) [reply appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java]: What does it add for semantics knowing that the method comes from the `Collections ` class?

What you need to know reading the code is that a `singletonMap` is returned. Is there also another type of `singletonMap` that is not a `Collection`?

Comment (Cousjava) [reply appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java]: By just calling `singletonMap()` I would presume that is was directly present in the class or a superclass, it is non-obvious that it comes from the `Collections` class.

Comment (arjantijms) [reply appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java]: If it's only about "where it comes from" and not "the meaning itself isn't clear", then it doesn't matter ;)

**Comment by arjantijms** [comment]
Jenkins test please

**Comment by Cousjava** [comment]
Jenkins test please

File: appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java (hunk 2 above)
Comment (Cousjava) [reply appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/MetricsService.java]: I would still say it isn't clear is a local method could well do data processing, which the one in the Collections framework does not.

**Pull request merged**